src/nlua_gfx.c

```diff
@@ -467,29 +467,28 @@ static int gfxL_renderTexH( lua_State *L )
  *    @luatparam number w Width of the rectangle.
  *    @luatparam number h Height of the rectangle.
  *    @luatparam Colour col Colour to use.
- *    @luatparam[opt=false] boolean empty Whether or not it should be empty.
+ *    @luatparam[opt=0] int lw Width of the outline or 0 to fill.
+ *    @luatparam[opt=0] int rx The corners curve length along x axis.
+ *    @luatparam[opt=0] int ry The corners curve length along y axis.
  * @luafunc renderRect
  */
 static int gfxL_renderRect( lua_State *L )
 {
    glColour *col;
-   double    x, y, w, h;
-   int       empty;
+   double    x, y, w, h, lw, rx, ry;
 
    /* Parse parameters. */
-   x     = luaL_checknumber( L, 1 );
-   y     = luaL_checknumber( L, 2 );
-   w     = luaL_checknumber( L, 3 );
-   h     = luaL_checknumber( L, 4 );
-   col   = luaL_checkcolour( L, 5 );
-   empty = lua_toboolean( L, 6 );
+   x   = luaL_checknumber( L, 1 );
+   y   = luaL_checknumber( L, 2 );
+   w   = luaL_checknumber( L, 3 );
+   h   = luaL_checknumber( L, 4 );
+   col = luaL_checkcolour( L, 5 );
+   lw  = lua_tonumber( L, 6 );
```

Comment:
If you want to make it optional, you have to use `luaL_optnumber( L, 6, 0. );` where `0.` would be the default in this case. This will raise errors if non-numbers are passed.